### DISCUSSION TITLE
Possibility of separating the application interface from the editor core

### DISCUSSION THREAD

**Discussion by douglasmatheus** [Hufe921/canvas-editor#641] [Ideas]
Hello, I would like to discuss the possibility of separating the application interface from the editor core.

As the main focus of this project is the editor layer itself and not the app mode. This is interesting because this way we can have a project focused only on application mode, develop more responsiveness for the menus and other components, in addition to implementing and improving features that are more specific to application mode, for example, comments, change history, CRDT (I believe that the editor core needs to implement the multiple cursors feature), margin ruler, etc.

I am raising this discussion to understand and be sure of what the author intends as the direction of the project. And in my case, I use this project (both application mode and the editor core in my projects), so certain features that I implemented on my own need to always be repeated each time I clone the project, for example internationalization of the application menus .

I did some tests here with this same application, simply removing the src/editor directory and installing the editor via npm as instructed in the documentation and everything worked perfectly.

@Hufe921  could at least consider creating another repository from this one with just the application part.

**Comment by Hufe921** [comment]
The editor will be transformed into a monorepo project when its functionality is perfected
